README.md

```diff
@@ -322,10 +424,75 @@ key = "CREATED" # options: "CREATED"|"MODIFIED"
+results = client.search(record_and_plain_search)
+# The results we see here is all records of record_type `flora`, and key `*test*` and value `*dand*`
+# We get the single record with meta "{'flora_test_12345':'dandelion'}"
+print_results("get records by record type 'flora' AND plain containing '*test*':'*dand*' ", results)
+```
+
+Combining match and exclude together will remove the exclude fields 
+```python
+# This means records that equal match terms AND do not equal exclude terms are returned.
+match_and_exclude = Search().match(record_types=["flora"]).exclude(strategy="WILDCARD", keys=["*test*"])
+results = client.search(match_and_exclude)
+print_results("get records of record type 'flora' and do not have keys `*test*`", results)
+```
+
+Chaining the match methods with match, and exclude methods with excludes are joined by the logical OR. In general, the two basic tenants of chaining these methods are as follows:
+1. more match calls expands your search, adding fields that match to your results
+1. more exclude callscontstricts your search, removing fields that match from your results
+```python
+# Chaining match terms with match or exclude terms with exclude are joined by the logical OR
+chain_match = Search().match(condition="AND", record_types=["flora"]).match(condition="AND", record_types=["fauna"])
+# the above search is equivalent to below.
+equivalent_to_chain_match = Search().match(condition="OR", record_types=["flora", "fauna"])
+```
+
+Nested chaining allows you to specify varying strategies for different fields 
+```python
+differing_strategies = Search().match(strategy="EXACT", record_types=["flora"])\
+                                .match(strategy="WILDCARD", keys=["*12345"])\
+                                .exclude(strategy="REGEXP", keys=[".*test.*"])\
+                                .exclude(strategy="REGEXP", keys=[".*server_2.*"])
+# results = (MATCH `flora` OR MATCH `*12345`) AND (EXCLUDE `.*test.*` OR EXCLUDE `.*server_2.*`)
+results = client.search(differing_strategies)
+print_results("Different matching strategies: this search will return an EXACT match to record_type `flora` OR WILDCARD match to keys `*12345`, and a REGEXP exclude to keys `.*test.*` OR REGEXP exclude to keys `.*server_2.*`", results)
+```
+
+Keep in mind that this chaining means your previous search object gets altered each time. 
+```python
+original_search = Search().exclude(record_types=["fauna"])
+modified_search = original_search.exclude(record_types=["flora"])
+results = client.search(modified_search)
+print_results("modified_search and original_search will exclude both flora and fauna", results)
+```
+
 ### Advanced Matching Strategies
 
 Search offers advanced queries that provide more flexibility than the default matching strategy of `EXACT`. The four options are `EXACT`, `FUZZY`, `WILDCARD`, and `REGEXP`. Depending on the matching strategy and the fields provided, these searches may be slower. 
```

Comment:
Might be good to document the default ordering of performance here from faster to slower: `EXACT > FUZZY > WILIDCARD > REGEXP`